src/proxy-animation.js

```diff
@@ -30,151 +30,470 @@ function createReadyPromise(details) {
+    const playbackRate = effectivePlaybackRate(details);
+    const pending = this.pending;
+
+    if (fromScrollTimeline) {
+      removeAnimation(details.timeline, details.animation);
+    }
+
+    details.resetCurrentTimeOnResume = false;
+    if (toScrollTimeline) {
+      // Cannot assume that the underlying native implementation supports
+      // mutable timelines. Thus, we leave its timeline untouched, and simply
+      // ensure that it is in the paused state.
+      details.timeline = newTimeline;
+      applyPendingPlaybackRate(details);
+      details.animation.pause();
+      switch(previousPlayState) {
+        case 'idle':
+          details.playState = 'idle';
+          details.holdTime = null;
+          details.startTime = null;
+          break;
+
+        case 'paused':
+          details.playState = 'paused';
+          details.resetCurrentTimeOnResume = true;
+          details.animation.currentTime = previousCurrentTime;
+          break;
+
+        case 'running':
+        case 'finished':
+          details.playState = 'running';
+          details.startTime =
+              playbackRate < 0 ? details.animation.effect.getTiming().duration
+                               : 0;
+          details.holdTime = null;
+          break;
+      }
+      addAnimation(details.timeline, this);
+      if (pending)
+        createReadyPromise(details);
+    } else {
+      // TODO: polyfill mutable timeline support. Cannot assume the native
+      // animation supports mutable timelines. Could keep a list of detached
+      // timelines and pump updates to current time via rAF.
+      details.animation.timeline = newTimeline;
+      if (fromScrollTimeline) {
+        // TODO: sync pending status & play state (ready promise).
+        details.timeline = null;
+        details.animation.currentTime = previousCurrentTime;
+      }
+    }
   }
 
   get startTime() {
-    const internalPlayState = proxyAnimations.get(this).playState;
-    if (!internalPlayState)
-      return proxyAnimations.get(this).animation.startTime;
-    if (internalPlayState == "running")
-      return 0;
-    return null;
+    return getAnimationProperty(proxyAnimations.get(this), 'startTime');
   }
   set startTime(value) {
-    proxyAnimations.get(this).animation.startTime = value;
+    const previousCurrentTime = this.currentTime;
+    const details = proxyAnimations.get(this);
+
+    if (!details.timeline) {
+      details.animation.startTime = value;
+      return;
+    }
+
+    details.resetCurrentTimeOnResume = false;
+    applyPendingPlaybackRate(details);
+    details.readyPromise = null;
+    if (typeof value == 'number') {
+      details.holdTime = null;
+      details.startTime = value;
+      details.playState = 'running';
+      const timelineTime = details.timeline.currentTime;
+      details.animation.currentTime =
+          (timelineTime - details.startTime) * this.playbackRate;
+      updateFinishedState(details);
+    } else {
```

Comment:
What does not 'number' mean? undefined? null? can we explicitly test for this value with === rather than consulting the type and throw if we get the wrong type?